fenix/app/src/main/java/org/mozilla/fenix/translations/automatic/AutomaticTranslation.kt

```diff
@@ -0,0 +1,117 @@
+
+package org.mozilla.fenix.translations.automatic
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.heading
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.list.TextListItem
+import org.mozilla.fenix.theme.FirefoxTheme
+import java.util.Locale
+
+/**
+ * Firefox Automatic Translate fragment compose view.
+ *
+ * @param automaticTranslationList List of [AutomaticTranslationItem]s to display.
+ * @param onItemClick Invoked when the user clicks on the a item from the list.
+ */
+@Composable
+fun AutomaticTranslation(
+    automaticTranslationList: List<AutomaticTranslationItem>,
+    onItemClick: (AutomaticTranslationItem) -> Unit,
+) {
+    Column(
+        modifier = Modifier
+            .background(
+                color = FirefoxTheme.colors.layer1,
+            ),
+    ) {
+        Text(
+            text = stringResource(R.string.automatic_translation_header),
+            modifier = Modifier
+                .padding(
+                    start = 72.dp,
+                    top = 6.dp,
+                    bottom = 6.dp,
+                    end = 16.dp,
+                )
+                .semantics { heading() },
+            color = FirefoxTheme.colors.textPrimary,
+            style = FirefoxTheme.typography.subtitle1,
+        )
+
+        LazyColumn {
+            items(automaticTranslationList) { item: AutomaticTranslationItem ->
+                var description: String? = null
+                if (item.automaticTranslationOption !is AutomaticTranslationOption.OfferToTranslate) {
+                    description = stringResource(item.automaticTranslationOption.titleId)
+                }
+                TextListItem(
+                    label = item.languageLocale.displayLanguage,
+                    description = description,
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(start = 56.dp),
+                    onClick = {
+                        onItemClick(item)
+                    },
+                )
+            }
+        }
+    }
+}
+
+@Composable
+internal fun getAutomaticTranslationList(): List<AutomaticTranslationItem> {
+    return mutableListOf<AutomaticTranslationItem>().apply {
+        add(
+            AutomaticTranslationItem(
+                languageLocale = Locale.CANADA,
```

Comment:
Thanks for adding the placeholder data!

This area will change when we do integration into AC, but I'd prefer if our placeholder data was a bit more similar. 

**Background:**
Our GV API calls here will likely be [`getLanguageSettings()`](https://searchfox.org/mozilla-central/rev/42acdc9cd5ae89222bdceeeaed7bacac755be48f/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/TranslationsController.java#239) and [`listSupportedLanguages()`](https://searchfox.org/mozilla-central/rev/42acdc9cd5ae89222bdceeeaed7bacac755be48f/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/TranslationsController.java#133). I'm still thinking through this in AC, but we will likely call something that makes these calls more convenient, something like:

`List<AutomaticTranslationPreference> getAutomaticTranslationPrefs` (Will be a new AC call that coordinates the two GV calls.):

`AutomaticTranslationPreference`:
* [`Language`](https://searchfox.org/mozilla-central/rev/42acdc9cd5ae89222bdceeeaed7bacac755be48f/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/TranslationsController.java#1043):
  * `code` - String BCP 47 code of the language
  * `localizedDislayName` - String language name
* [`LanguageSetting`](https://searchfox.org/mozilla-central/rev/42acdc9cd5ae89222bdceeeaed7bacac755be48f/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/TranslationsController.java#549-567) - enum 

**Requested Change:**
* Change `languageLocale` to `displayName` (For the placeholder data, just make the `Locale.displayLanguage` call here)
  * (This placeholder will ultimately probably become `Language`, which is about to be introduced as soon as this [patch](https://github.com/mozilla-mobile/firefox-android/pull/4521/files) finishes review.)
* (More of a thought or item to look at): Not sure best how to orcestrate it, but when we are ready to commit the change the user made (e.g., offer, always, never), we will need to call [`setLanguageSettings`](https://searchfox.org/mozilla-central/rev/42acdc9cd5ae89222bdceeeaed7bacac755be48f/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/TranslationsController.java#278). So, at the point of commiting the change to GV, we will need to know the BCP 47 `code` and the radio `LanguageSetting` enum to use. (What are your thoughts?)

Reply:
Done